src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -180,9 +198,7 @@ internal static bool IsOperationOfAsyncPageable(CSharpType type)
         internal static bool IsOperationOfPageable(CSharpType type)
             => type.IsFrameworkType && type.FrameworkType == typeof(Operation<>) && type.Arguments.Length == 1 && IsPageable(type.Arguments[0]);
 
-        internal static Type? ToFrameworkType(Schema schema) => ToFrameworkType(schema, schema.Extensions);
-
-        internal static Type? ToFrameworkType(Schema schema, RecordOfStringAndAny? extensions) => schema.Type switch
```

Comment:
This combination also needs your attention @ArthurMa1978

Reply:
yes, for parameter the extension is not under scheme but parameter, so here don't merge back

Reply:
Fixed

Reply:
I find this issue is not fixed actually. In some cases (maybe in parameters?) if we add an extension to that parameter (string typed), the extension will not show up in the extension of that corresponding schema, instead, it will show up in the extension of the parameter (RequestParameter from code model). This is the reason that we need a overload of this method to pass in an extension dictionary.
We have similar case for method `CreateType`